### PR TITLE
docs: Updates for versioned secrets and calling out config tools

### PR DESCRIPTION
**Pull request by jkayani**
### Description

Docs for feature introduced in #189 

Also moving the docs about Helm, Kustomize, etc. to a more prominent location since they seem to be a bit hidden in their current location. Might help avoid confusion like in #178 

### Checklist
Please make sure that your PR fulfills the following requirements:
- [ ] Reviewed the guidelines for contributing to this repository
- [ ] The commit message follows the [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] Tests for the changes have been updated
- [ ] Docs have been added / updated

### Type of Change

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [x] Documentation content changes
- [ ] Other (please describe)

### CODE REVIEW

File: docs/config.md [hunk 1]
```diff
@@ -44,14 +44,15 @@ We support all Vault Environment Variables listed [here](https://www.vaultprojec
 | --------------- | ----------- | ----- |
 | AVP_TYPE           | The type of Vault backend  | Supported values: `vault`, `ibmsecretsmanager`, `awssecretsmanager` and `gcpsecretmanager` |
 | AVP_KV_VERSION    | The vault secret engine  | Supported values: `1` and `2` (defaults to 2). KV_VERSION will be ignored if the `avp.kubernetes.io/kv-version` annotation is present in a YAML resource.|
-| AVP_AUTH_TYPE      | The type of authentication | Supported values: vault: `approle, github, k8s`   ibmsecretsmanager: `iam` |
+| AVP_AUTH_TYPE      | The type of authentication | Supported values: vault: `approle, github, k8s` |
```
Comment (werne2j) [inline docs/config.md:47]: Maybe change the Notes to say something about `only used for a Vault backend`

**Review by werne2j — APPROVED** [review]
Just the one small thing

**Comment by jkayani** [comment]
Going to try to re-work the "How it works" page to be clear on all the different behaviors

File: docs/usage.md [hunk 2]
```diff
@@ -37,12 +37,71 @@ spec:
 3. Or you can pass the config-management-plugin flag to the Argo CD CLI app create command:  
 `argocd app create you-app-name --config-management-plugin argocd-vault-plugin`
 
+#### With `helm`
```
Comment (werne2j) [inline docs/usage.md:40]: Lets not do helm, kustomize or jsonnet with the back ticks, i think it throws off the formatting

**Review by werne2j — APPROVED** [review]
Looks good, one small suggestiong

**Pull request merged**